### PR TITLE
Climbing stairs

### PR DESCRIPTION
**Pull request by Shweta2024**
## Related Issue

-Climbing Stairs - using Dynamic Programming #993
#### Climbing Stairs:-
You are climbing a staircase. It takes n steps to reach the top.

Each time you can either climb 1 or 2 steps. In how many distinct ways can you climb to the top?

## Proposed Changes
I have added the following things: -
-  Problem Statement
-  Examples
-  Explanation
-  Code
-  Time Complexity & Space Complexity

## Additional Info
- [Link of the problem](https://leetcode.com/problems/climbing-stairs/)

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

#### Climbing Stairs

You are climbing a staircase. It takes n steps to reach the top.

Each time you can either climb 1 or 2 steps. In how many distinct ways can you climb to the top?

We were supposed to find the total number of ways in which we can reach the n-th step, provided we can either make one jump or 2 jumps at a time.

I have added the Problem Statement and Examples along with the Explanation on how to approach this problem, a commented Code, and also written the Time Complexity and Space Complexity. And finally updated the README.md.

This change helps to solve a very famous problem of Dynamic Programming i.e. Climbing Stairs and this problem helps in understanding the concept of Dynamic Programming.

I coded this problem in VS Code and tested the given problem on the test cases in the range of 1 to 44. And it is working fine for them.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Screenshot of Output when the given input is 4|  Screenshot of Output when the given input is 44  |
| op1 | op2
 |

### CODE REVIEW

File: D-DynamicProgramming/README.md [hunk 1]
```diff
@@ -403,3 +405,80 @@ Thus it takes 14 trials to test a 100-floor building.
...
+
+- [Analysis](#analysis)
+
+### Problem Statement of Climbing Stairs
+
+You are climbing a staircase. It takes n steps to reach the top.
+
+Each time you can either climb 1 or 2 steps. In how many distinct ways can you climb to the top?
+
+
```
Comment (Lakhankumawat) [inline D-DynamicProgramming/README.md:429]: Add a working image for your problem just like others see README for more .


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Comment by Lakhankumawat** [comment]
Where is your issue number ?

**Comment by Lakhankumawat** [comment]
Every PR is linked with an issue

**Comment by Shweta2024** [comment]
> Where is your issue number?

Sir, I have added the issue number please check.
Thank you

**Review by Lakhankumawat — APPROVED** [review]
LGTM ✨🎉

Good work @Shweta2024  , keep contributing keep shining

**Comment by Lakhankumawat** [comment]
@all-contributors please add @Shweta2024  code ,

**Comment by Shweta2024** [comment]
> 

Thank you sir

**Pull request merged**